src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -143,6 +142,20 @@ private void VerifyExpression(BoundExpression expression, bool overrideSkippedEx
                 return null;
             }
 
+            public override BoundNode? VisitAssignmentOperator(BoundAssignmentOperator node)
+            {
+                // We're not correctly visiting the right side of object creation initializers when
+                // the symbol is null (such as for dynamic)
+                // https://github.com/dotnet/roslyn/issues/45088
+                if (node.Left is BoundObjectInitializerMember { MemberSymbol: null })
+                {
+                    VerifyExpression(node);
+                    return Visit(node.Left);
```

Comment:
>return Visit(node.Left); [](start = 20, length = 24)

This looks suspicious. Why are we replacing an assignment with its left side? If return value is not important, let's return null.